FreeRTOS/FreeRTOS
# Improve connection sharing demo

Description
-----------

Makes changes to the connection sharing demo after feedback. Changes include:
* Add a daemon socket listener so that the process loop will only execute if data is available on the socket.
* Set process loop timeout to 0 so that other commands do not have to wait for a process loop that isn't doing anything.
* Creates an additional publisher task to show multiple tasks publishing simultaneously.
* Clarify what is meant by "synchronous" and "asynchronous" publishes.
* Fix possible race condition in task notification waits.
* Move task notification wait loops to its own function.
* The `prvMQTTConnect` function was doing more than just connecting, so it was split into more functions.
* Minor name changes for clarity.

Test Steps
-----------

Tested locally against mosquitto and the AWS IoT Broker.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
